doc/chapters/api/index.rst

```diff
@@ -24,88 +13,276 @@ Methods
 
-void ShutdownContainerWithTimeout(uint32 containerID, uint32 timeout)
 
 WriteToStdIn
 ------------
-:processID: uint32
-:bytes: array<char>
+Send characters array to the standard input of particular process.
+
+:Parameters:
+        - **processID** ``uint32`` PID of the process; obtained LaunchCommand.
+        - **bytes** ``array<char>`` character array to sent stdin.
+
+:Return Value:
+        - **None**
+
+:Example Usage:
+
+::
+
+        bus-send --system --print-reply \
+        --dest=com.pelagicore.SoftwareContainerAgent \
+        /com/pelagicore/SoftwareContainerAgent \
+        com.pelagicore.SoftwareContainerAgent.WriteToStdIn \
+        uint32:14859 \
+        array:byte:'a','b'        
 
-void WriteToStdIn(uint32 processID, array<char> bytes)
 
 BindMountFolderInContainer
 --------------------------
-:containerID: uint32
-:pathInHost: string
-:subPathInContainer: string
-:readOnly: bool
+Binds a directory in the container to a directory in the host/
+
+:Parameters:
+        - **containerID** ``uint32`` The ID obtained by CreateContainer method.
+        - **pathInHost** ``string`` path to the directory in host.
+        - **subPathInContainer** ``string`` path to the directory in container.
+        - **readOnly** ``bool`` availability of binded folder 
+
+:Return Value:
+        - **pathInContainer** ``string`` path to the bind folder in container. 
 
-string pathInContainer BindMountFolderInContainer(uint32 containerID, string pathInHost, string subPathInContainer, ool readOnly)
+
+:Example Usage:
+
+::
+
+        dbus-send --system --print-reply \
+        --dest=com.pelagicore.SoftwareContainerAgent \
+        /com/pelagicore/SoftwareContainerAgent \
+        com.pelagicore.SoftwareContainerAgent.BindMountFolderInContainer \
+        uint32:1 \
+        string:"/home/myUser/myBindFolder" \
+        string:"/home/vagrant/softwarecontainer/build" \
+        boolean:false
+        
 
 SetGatewayConfigs
 -----------------
-:containerID: uint32
-:configs: map<string, string>
+Sets the configuration of particular gateway. The gateway configuration contains settings as key/value pairs.
+
+:Parameters:
+        - **containerID** ``uint32`` The ID obtained by CreateContainer method.
+        - **configs** ``map<string, string>`` A map to key/value pairs.
+
+:Return Value:
+       - **None**
+
+ 
+:Example Usage:
+
+::
+
+        dbus-send --system --print-reply \
+        --dest=com.pelagicore.SoftwareContainerAgent \
+        /com/pelagicore/SoftwareContainerAgent \
+        com.pelagicore.SoftwareContainerAgent.SetGatewayConfigs \
+        uint32:1 \
+        dict:string:string:"configItem1","1","configItem2","2"
 
-void SetGatewayConfigs(uint32 containerID, map<string, string> configs)
 
 SetCapabilities
 ---------------
-:containerID: uint32
-:capabilities: array<string>
+Currently This method has no applicable usage. 
+
+:Parameters:
+        - **containerID** ``uint32`` The ID obtained by CreateContainer method.
```

Comment:
Actually... I just realized, it's good to know where the ID comes from, but would also be interesting to know what it is... Kind of obvious I guess, but none the less..